### DISCUSSION TITLE
Golang module replace directive equivalent in V?

### DISCUSSION THREAD

**Discussion by jorgeluismireles** [vlang/v#22127] [Questions and Answers]
### My problem
I started a golang project with several gitlab repos, some libraries and some apps. Since all the repos were **public** any repo could be imported as a library as usual. From app `server.go`:
```
package server
import (
   gitlab.com/username/protocol/defs
  ...
)
```
And in server repo we have a `go.mod` like this:
```
module gitlab.com/username/server
go 1.22.2
require (
    gitlab.com/username/protocol v0.0.0-2024-etc-etc
)
```
Then my employer request me to push all the code to **private** repositories so libraries could not be imported directly with `go get` or `get mod tidy`.

### My solution
I finally use the *replace* directive in server app `go.mod` as follows:
```
module gitlab.com/username/server
go 1.22.2
replace (
    gitlab.com/username/protocol => ../protocol
)
require (
	gitlab.com/username/protocol v0.0.0
)
```
To use *replace*, the library now should be read from the filesystem and should be marked as version v0.0.0 (in require directive). This works *offline* if your code at your station is contigous:
```
gitlab.com
  '- username
       |- protocol
       |- server
       |    '- go.mod <- with replace directive   
       '- client
```

### Question 1
What approach is recommended in big V projects and with **private** repos?

### Question 2
In go if your repo is called `gitlab.com/username/protocol-inputs` and you imported, within the code you could access the path with only *inputs* without *protocol* part, as an automatic alias. Example:
```
package server
import "gitlab.com/username/protocol-inputs"
func main() {
     inputs.Debug() // inputs not protocol-inputs
}
```
Similarly if your repo is defined as `github.com/gdamore/tcell/v2 v2.6.0` it'll be accessible only with `tcell` without any version.
Is an equivalent of modules stripped by dashes and versions?

Thanks in advance.

**Comment by ctkjose** [comment]
In V modules and packages are pretty much the same. They are more straight forward, you can organize your project in folders and just use them as modules. Read more here: https://v-lang.dev
